### ISSUE TITLE
Xorg crash on 2nd suspend/resume cycle with Optimus active

### ISSUE THREAD

**Issue by JonnyMako** [Bumblebee-Project/Bumblebee#745]
I'm using 4.4.3-1-ARCH and bumblebee 3.2.1-10 with bbswitch 0.8-42.

I have a weird problem. With intel-virtual-output active, if I suspend and resume, my Xorg crashes on the second suspend/resume cycle. I am unable to get useful logs for the crash.

My bumblebee.conf is unchanged and relevant options in xorg.conf.nvidia are:

Option "ConnectedMonitor" "DFP"
Option "UseDisplayDevice" "DFP"

Can someone help me find the culprit here? Thanks!

(I manually startx from tty1 at login)

**Comment by JonnyMako** [comment]
**I was finally able to get a backtrace in the log:**
`[    34.206] (II) systemd-logind: got resume for 13:69`
`[    34.220] (II) systemd-logind: got resume for 13:81`
`[    34.233] (II) systemd-logind: got resume for 13:82`
`[    34.246] (II) systemd-logind: got resume for 13:79`
`[    34.247] (II) systemd-logind: got resume for 226:0`
`[    34.247] (II) Open ACPI successful (/var/run/acpid.socket)`
`[    34.247] (II) AIGLX: Resuming AIGLX clients after VT switch`
`[    34.247] (II) intel(0): switch to mode 1920x1080@60.0 on eDP1 using pipe 0, position (0, 0),` `rotation normal, reflection none`
`[    34.262] (EE)`
`[    34.262] (EE) Backtrace:`
`[    34.262] (EE) 0: /usr/lib/xorg-server/Xorg (OsLookupColor+0x139) [0x598b99]`
`[    34.265] (EE) 1: /usr/lib/libc.so.6 (__restore_rt+0x0) [0x7fd76e0b032f]`
`[    34.265] (EE) 2: /usr/lib/libc.so.6 (malloc_consolidate+0x112) [0x7fd76e0f1f62]`
`[    34.266] (EE) 3: /usr/lib/libc.so.6 (_int_malloc+0x160) [0x7fd76e0f3c90]`
`[    34.266] (EE) 4: /usr/lib/libc.so.6 (__libc_malloc+0x54) [0x7fd76e0f5924]`
`[    34.266] (EE) 5: /usr/lib/xorg-server/Xorg (xf86CrtcRotate+0x749) [0x4bc939]`
`[    34.266] (EE) 6: /usr/lib/xorg-server/Xorg (xf86DiDGAReInit+0x58) [0x4bc4c8]`
`[    34.266] (EE) 7: /usr/lib/xorg-server/Xorg (xf86PruneDuplicateModes+0x13ff) [0x4b9d2f]`
`[    34.267] (EE) 8: /usr/lib/xorg-server/Xorg (RRGetInfo+0xeb) [0x4fa56b]`
`[    34.267] (EE) 9: /usr/lib/xorg/modules/extensions/libglx.so (GlxSetVisualConfigs+0x65ff)` `[0x7fd76baaa17f]`
`[    34.267] (EE) 10: /usr/lib/xorg-server/Xorg (xf86VTEnter+0x88) [0x4773b8]`
`[    34.267] (EE) 11: /usr/lib/xorg-server/Xorg (xf86SIGIOSupported+0x867) [0x4a1447]`
`[    34.267] (EE) 12: /usr/lib/xorg-server/Xorg (xf86SIGIOSupported+0xb6b) [0x4a17fb]`
`[    34.267] (EE) 13: /usr/lib/libdbus-1.so.3 (dbus_connection_dispatch+0x375) [0x7fd76fd45085]`
`[    34.268] (EE) 14: /usr/lib/libdbus-1.so.3 (dbus_connection_dispatch+0x6dd) [0x7fd76fd45a2d]`
`[    34.268] (EE) 15: /usr/lib/xorg-server/Xorg (config_fini+0x4c1) [0x49a501]`
`[    34.268] (EE) 16: /usr/lib/xorg-server/Xorg (WakeupHandler+0x6d) [0x43ae0d]`
`[    34.268] (EE) 17: /usr/lib/xorg-server/Xorg (WaitForSomething+0x1ef) [0x59139f]`
`[    34.268] (EE) 18: /usr/lib/xorg-server/Xorg (SendErrorToClient+0x10e) [0x43611e]`
`[    34.268] (EE) 19: /usr/lib/xorg-server/Xorg (remove_fs_handlers+0x453) [0x43a313]`
`[    34.268] (EE) 20: /usr/lib/libc.so.6 (__libc_start_main+0xf0) [0x7fd76e09d710]`
`[    34.269] (EE) 21: /usr/lib/xorg-server/Xorg (_start+0x29) [0x424649]`
`[    34.269] (EE) 22: ? (?+0x29) [0x29]`
`[    34.269] (EE)`
`[    34.269] (EE) Segmentation fault at address 0x7fd76c7d3c98`
`[    34.269] (EE)`
`Fatal server error:`
`[    34.269] (EE) Caught signal 11 (Segmentation fault). Server aborting`
`[    34.269] (EE)`
`[    34.269] (EE)`

**Comment by teozkr** [comment]
This also seems to be happening after just switching to a tty (ctrl+alt+#N) and then back again.

My workaround atm is the following systemd target, which unloads the nvidia drivers as part of the suspend process:

```systemd
[Unit]
Before=suspend.target

[Service]
Type=simple
ExecStart=/bin/modprobe -r nvidia-modeset

[Install]
WantedBy=suspend.target
```

My guess would be that this is a two-part problem: the nvidia driver crashes upon resuming/switching, and Bumblebee doesn't actually unload the module correctly when it's no longer in use.

**Comment by teozkr** [comment]
I'd guess https://devtalk.nvidia.com/default/topic/925979/ctl-alt-f7-and-wake-up-from-suspend-not-working-with-nvidia-364-12-/ is the same issue as well.

**Comment by ArchangeGabriel** [comment]
Indeed, there is an issue with unloading (#719, should be solved by #762). For the suspend problem, I’ve seen other reports, will need to dig into it a bit.
